### PR TITLE
fix: REST API filter by UUID bugfixes

### PR DESCRIPTION
**Pull request by IvanJov**
#### Short description of what this resolves:
Dear Community,  I am representing the project of a Digital health platform developed by Cardo Health AB ( [www.cardohealth.com](http://www.cardohealth.com/)),  through the support of my current employer ( NearForm ). We are pleased to contribute some fixes in OpenEMR on behalf of Cardo Health and look forward to continuing to do so in the future.

In this PR there are multiple fixes:
- filtering by UUID was broken on some API endpoints
- filtering in general was broken because of a wrong usage of `$search` variable instead of `$newSearch`
- some API endpoints were returning 500 error because byte UUID from DB wasn't converted to string before returning it in a REST API response

### CODE REVIEW

**Comment by adunsulag** [comment]
So I just ran this PR through the ONC Inferno FHIR test suite and the Condition endpoint is broken for FHIR with this PR.

<fhirUrl>/apis/default/fhir/Condition?patient=<uuid> will return a 500 error.

File: src/Services/ConditionService.php [hunk 1]
```diff
@@ -119,7 +119,11 @@ public function createResultRecordFromDatabaseResult($row)
      */
     public function getAll($search = array(), $isAndCondition = true, $puuidBind = null)
     {
-        $newSearch = [];
```
Comment (adunsulag) [inline src/Services/ConditionService.php:122]: Why remove this line?  $newSearch still needs to be defined.

Comment (adunsulag) [reply src/Services/ConditionService.php:122]: This is probably why the condition is failing in the test suite.  Everything else looked fine.

Comment (IvanJov) [reply src/Services/ConditionService.php:122]: Oops, somehow this got in. Fixed 👍

**Comment by IvanJov** [comment]
@adunsulag Thanks for the review! Pushed change. Can you rerun the test suite? Thanks!

File: src/Services/ConditionService.php [hunk 2]
```diff
@@ -120,6 +120,12 @@ public function createResultRecordFromDatabaseResult($row)
     public function getAll($search = array(), $isAndCondition = true, $puuidBind = null)
     {
         $newSearch = [];
+
+        // override puuid with the token search field for binary search
+        if (isset($search)) {
+            $newSearch['puuid'] = new TokenSearchField('puuid', $search, true);
```
Comment (adunsulag) [inline src/Services/ConditionService.php:126]: So this is still failing in Inferno.  I dug into the error and its failing here on the TokenSearchField.  I don't think you are wanting to send in the entire key/value search array in as values right?  Shouldn't this be something like this?
```
$newSearch['puuid'] = new TokenSearchField('puuid', $search['patient'], true); 
```
Here is the error log from the Inferno test (Note you'll have to scroll to the right to see the details):
```
[Thu Feb 02 21:27:33.729803 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.728676-05:00] OpenEMR.DEBUG: dispatch.php authenticating user [] []
[Thu Feb 02 21:27:33.744571 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.744486-05:00] OpenEMR.DEBUG: dispatch.php valid role and user has access to api/fhir resource {"resource":"/fhir/Condition"} []
[Thu Feb 02 21:27:33.745050 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.744981-05:00] OpenEMR.DEBUG: dispatch.php request setup for user role {"authUserID":"5","authUser":"aldavis"} []
[Thu Feb 02 21:27:33.747049 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.746984-05:00] OpenEMR.DEBUG: dispatch.php api is userRole populating token context for request due to smart launch scope [] []
[Thu Feb 02 21:27:33.749026 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.748951-05:00] OpenEMR.DEBUG: HttpRestRouteHandler::dispatch() start request {"resource":"","method":"GET","user":"","role":"users","client":"mRbtu9Uca2mfZZoNEBXJumnKgjUMK4ySPWWaUu5Q0V4","apiType":"fhir","route":"/fhir/Condition","queryParams":[]} []
[Thu Feb 02 21:27:33.749447 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.749383-05:00] OpenEMR.DEBUG: HttpRestParsedRoute->__construct() matched {"routePath":"GET /fhir/Condition","requestPath":"/fhir/Condition","method":"GET","routeParams":[],"resource":"Condition","operation":""} []
[Thu Feb 02 21:27:33.749543 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.749501-05:00] OpenEMR.DEBUG: checkSecurity() - patient specific request, so only allowing access to records to that one patient [] []
[Thu Feb 02 21:27:33.749634 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.749591-05:00] OpenEMR.DEBUG: HttpRestRouteHandler->dispatch() dispatching route {"route":"GET /fhir/Condition"} []
[Thu Feb 02 21:27:33.777825 2023] [php:notice] [pid 1885] [client 66.190.181.57:50066] [2023-02-02T16:27:33.777748-05:00] OpenEMR.DEBUG: FhirServiceBase->getAll() Created search parameters  {"searchParameters":["patient"]} []
[Thu Feb 02 21:27:33.780398 2023] [php:error] [pid 1885] [client 66.190.181.57:50066] PHP Fatal error:  Uncaught InvalidArgumentException: Token value must be a valid string in /var/www/localhost/htdocs/openemr/src/Services/Search/TokenSearchField.php:79\nStack trace:\n#0 /var/www/localhost/htdocs/openemr/src/Services/Search/TokenSearchField.php(71): OpenEMR\\Services\\Search\\TokenSearchField->createTokenSearchValue()\n#1 /var/www/localhost/htdocs/openemr/src/Services/Search/BasicSearchField.php(46): OpenEMR\\Services\\Search\\TokenSearchField->setValues()\n#2 /var/www/localhost/htdocs/openemr/src/Services/Search/TokenSearchField.php(26): OpenEMR\\Services\\Search\\BasicSearchField->__construct()\n#3 /var/www/localhost/htdocs/openemr/src/Services/ConditionService.php(126): OpenEMR\\Services\\Search\\TokenSearchField->__construct()\n#4 /var/www/localhost/htdocs/openemr/src/Services/FHIR/FhirConditionService.php(213): OpenEMR\\Services\\ConditionService->getAll()\n#5 /var/www/localhost/htdocs/openemr/src/Services/FHIR/FhirServiceBase.php(211): OpenEMR\\Services\\FHIR\\FhirConditionService->searchForOpenEMRRecords()\n#6 /var/www/localhost/htdocs/openemr/src/RestControllers/FHIR/FhirConditionRestController.php(52): OpenEMR\\Services\\FHIR\\FhirServiceBase->getAll()\n#7 /var/www/localhost/htdocs/openemr/_rest_routes.inc.php(7852): OpenEMR\\RestControllers\\FHIR\\FhirConditionRestController->getAll()\n#8 /var/www/localhost/htdocs/openemr/src/Common/Http/HttpRestRouteHandler.php(91): {closure}()\n#9 /var/www/localhost/htdocs/openemr/apis/dispatch.php(386): OpenEMR\\Common\\Http\\HttpRestRouteHandler::dispatch()\n#10 {main}\n  thrown in /var/www/localhost/htdocs/openemr/src/Services/Search/TokenSearchField.php on line 79
```

Comment (IvanJov) [reply src/Services/ConditionService.php]: You are right, let me change it and test again

**Comment by IvanJov** [comment]
@adunsulag Hello! I pushed a change. Can you try the test suite again? I think that it will work now 🍻

**Comment by rehmanatcardo** [comment]
@adunsulag ,  would be great to know if you have any update on @IvanJov's request. Thanks in advance!

**Comment by rehmanatcardo** [comment]
@adunsulag , can you please point us to someone who can help close it.

**Comment by adunsulag** [comment]
Going to dig deeper into this, but pulled the PR and many of the _search POST requests are breaking now in the Inferno test suite for our ONC certification:
image

Also Condition fails completely:
image

**Pull request merged**